storage/client.go

```diff
@@ -134,6 +134,20 @@ func resolveOptions(s *settings, opts ...storageOption) {
 	}
 }
 
+// callSettings is a helper for resolving storage options against the settings
+// in the conctext of an individual call. This is to ensure that client-level
+// default settings are not mutated by two different calls getting options.
+//
+// Example: s := callSettings(c.settings, opts...)
+func callSettings(defaults *settings, opts ...storageOption) *settings {
+	if defaults == nil {
+		return nil
+	}
+	cs := *defaults
```

Comment:
Add a comment that this copies the config (though not the pointer values a layer down such as `retry`).